extract/ext/office_extract.py

```
@@ -381,30 +382,40 @@ def extract_docx(filename, password_list, output_folder):
         of = olefile.OleFileIO(filename)
     except IOError:
         raise ValueError("Corrupted OLE Document")
+    password = None
 
-    if of.exists("WordDocument"):
-        # Cannot parse these files yet
-        raise ValueError("Legacy Word Document")
-
-    elif of.exists("EncryptionInfo") and of.exists("EncryptedPackage"):
+    # checks new versions of office docs .***x i.e. .xlsx, .docx, etc.
+    if of.exists("EncryptionInfo") and of.exists("EncryptedPackage"):
         metadata = parse_enc_info(of.openstream("EncryptionInfo"))
-
-        password = None
+        # From the provided passwords, check the password and break if it's correct
         for pass_try in password_list:
             if check_password(pass_try, metadata) is True:
                 password = pass_try
                 break
+    # open the file in read-binary, assign value to file variable
+    file = msoffcrypto.OfficeFile(open(filename, "rb"))
+    if not password:
+        # re: older versions, such as xls, doc, ppt
+        for password in password_list:
+            try:
+                # use the provided password, if correct, break.
+                file.load_key(password=password)
+                break
+            except Exception as e:
+                e_repr = repr(e)
+                if "Failed to verify password" in e_repr:
+                    continue
 
```

We should raise the exception if "Failed to verify password" is not in e_repr